### DISCUSSION TITLE
Can `brew upgrade` treat `auto_updates` and `version: latest` casks differently?

### DISCUSSION THREAD

**Discussion by ywwry66** [Homebrew/discussions#73] [Everyday usage] [answered]
Currently I use brew to manage `auto_updates` casks: I disable auto update inside those apps and use `brew upgrade -- greedy` to upgrade all formulae and casks.

Today, I installed a `version: latest` cask (`steam`, if this is any useful), and this breaks my old workflow. `steam` just got updated every time I run `brew upgrade -- greedy`, which is definitely annoying.

Do you think if it is worthwhile to add a flag to `brew upgrade` which only updates the `auto_updates` casks and leaves alone the `version: latest` casks? Or should I just remove `greedy` and turn on auto update functionality built into the apps?

**Comment by SMillerDev** [comment, marked as answer]
Yeah, definitely the later. That's what it's meant to do after all.

**Reply by ywwry66** [reply]
Thanks. May I ask what is the advantage of using app builtin auto update over using the `--greedy` flag? And under what circumstances could `--greedy` be useful?

**Reply by SMillerDev** [reply]
Generally the apps will try and update themselves whenever possible. Homebrew can't really avoid/fix that so it's better to just adapt to that reality. For myself there's some apps I rarely use but still get updates, so I run greedy for those every now and then.

**Reply by bmironer** [reply]
would it be possible to use the CFBundleShortVersionString in the Info.plist to get the currently installed version and compare with the cask ? 
better yet, do it from a function in the formula so authors can override it for the few casks with a custom version identification method ?

I can compare using 

`brew info --json=v2 --installed --casks | jq '.casks[] | {name, version, auto_updates, app:(.artifacts[].app) , outdated } | select(.auto_updates and .app) | { app:.app[0], version }'`

and 

`plutil -convert json -o - /Applications/*.app/Contents/Info.plist | jq '{CFBundleName, CFBundleShortVersionString}'`

But it's a bit hit and miss
